### DISCUSSION TITLE
Tags in media file

### DISCUSSION THREAD

**Discussion by ConcreteGarlic** [deckerst/aves#679] [General]
Hello - and thanks for the great software created here.

I am using Aves on daily basis on my mobile, though I need to move media from time to time to the PC.

Is there any hint - which image viewing software can be "compatibile" with Aves to view tags, created with Aves, and saved in jpg metadata? I did find Aves useful, as it does not create any external tags database - but what to use on PC, to not have all tags losts.

if we cant use here exact names of the pc software - if you can suggests please the name of feature I should look for in the pc software descriptions.

Cheers.
Garlic

**Comment by JanWaldhorn** [comment]
I use [Digikam ](https://www.digikam.org/) on the PC, but on Windows the Explorer can also show the tags under "Markers" or in the properties of the image.

**Comment by deckerst** [comment]
[Picasa](https://en.wikipedia.org/wiki/Picasa) (the discontinued product from Google) is still quality software that can read and write tags.

**Reply by totti4ever** [reply]
any chance to run it on a current macos?

**Reply by deckerst** [reply]
I don't know, and I'm not familiar with the mac ecosystem so i can't recommend anything.

**Comment by ConcreteGarlic** [comment]
Thanks for the replies. I am digging more into it :) 

Do I guess corectly (pure guess, honestly) that Aves saves the tags in IPTC metadata? So - in theory - any software that can read IPTC can do?
Usually, I use IrfanView for fast viewing, but my search now is for the app that can cooperate with Aves with proper by-tag sorting of the photo collection. 
If it is IPTC - it would be nice to find software that can "create" own categories based by incoming IPTC tags from new photos.

I will check both above suggestions. I installed XnViewMP few minutes ago - it does read IPTC, though it creates own database of sorting categories, at least as a default option. So not much my desired solution.

**Reply by satishguptajaipur** [reply]
I found eagle.cool the best software to do so. eagle.cool can automatically use the IPTC metadata as tags.
But I tried copying an image file tagged by Aves to eagle.cool and I couldn't find the tags assigned in Aves. Either I am not exporting the image properly or Aves is writing the data somewhere else.

I opened the same image in Affinity Photo and added some keywords in metadata named "File". Eagle.cool was able to use this metada as tags.

So I guess the problem is with Aves.

Also I wonder what happens if the phone crashes or I have to change the phone? Will my tags data be saved if I take backup somewhere and will Aves be able to read that data again on reinstall?

**Reply by deckerst** [reply]
I can't test `eagle.cool` because it's not available on Linux, but I'm surprised it does not see the tags edited in Aves. Tags are written in XMP, a standard more flexible and widespread than IPTC. As I've said below, Aves edits IPTC only if the section already exists, but it won't create it, preferring XMP over it.

**Comment by deckerst** [comment]
You can peruse https://github.com/deckerst/aves/wiki/Support:-Metadata-editing for details.

Basically, Exif and XMP is preferred. IPTC is modified only if it already exists in the file.

**Comment by kmcbest** [comment]
I'm on the same boat looking for an equivalent on Windows. 
First I've got FolderSync app to sync my neatly tagged images from my Android with a shared folder on PC. But sometimes I'm collecting images on PC too, FolderSync can sync them back to my Android, but I need a PC program that can tag like Aves. Otherwise I'll have to tag them on Aves and sync them back to my PC.

I tried XnView MP, it can read Aves tags and sort out images by tag. But its tag-writing mechanism isn't straightforward. Since XnView treat tags as "category" in its own glossary, if you apply a category to an image, it is registered in XnView database instead of written into the image's metadata. If you want to do so, you have to open the metadata dialog, type the tags (no suggestion either) and click "write", all those steps to achieve the same result Aves can do in three clicks.

Picasa's way of doing it looks off too. And the program really looks out of place by today's criteria.

**Comment by lordofscripts** [comment]
I didn't know they were written in the file, I will make it a point to just use it sparsely.

¿Is the modification date preserved after a tag is edited?

Is there a Linux version of Aves? I am using a Debian-based 3w-bit Raspberry π

**Comment by deckerst** [comment]
> Is the modification date preserved after a tag is edited?

No. In general you cannot trust Aves to preserve the file modification date, as Aves is a gallery app, not a file manager app, and does not have full access to file properties. 

> Is there a Linux version of Aves?

No there isn't. It could work inside a Waydroid container but I don't explicitly support this.

**Comment by kmcbest** [comment]
From my experience, after tagging a picture in Aves:
1. The picture's modification time is changed when seen in File manager (windows or Android) and will move to the top if you sort by most recent.
2. But if you have 'sort descendant by date' in Aves itself, the tagged picture won't move to the top, which is very carefully designed behavior knowing you may have to tag multiple times (otherwise you'll have great trouble finding the image again). So I think Aves sorts by creation date.

**Reply by deckerst** [reply]
> So I think Aves sorts by creation date.

It actually uses the metadata date when possible, falling back to the file modification date. The metadata date is typically from Exif, but it can be from XMP or other format specific field.

**Reply by lordofscripts** [reply]
And that can be easily solved by also programmatically changing the file modification timestamp with the EXIF Date Taken data.

**Reply by deckerst** [reply]
You assume that Aves can specify the `file modification date`, which is not possible. A gallery app, contrary to a general file manager, does not have full access to files, it only has a limited view of public media and can only modify them as such.
